### PR TITLE
docopts: migrate to buildGoModule

### PR DESCRIPTION
**Pull request by msanft**
## Description of changes

Migrating to `buildGoModule` so `buildGoPackage` can be deprecated as per https://github.com/NixOS/nixpkgs/issues/318069. Project needs an upstream patch to do so, as it does not have a vendor directory, nor is a module.

Also moved to `by-name`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/do/docopts/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+, fetchpatch
+}:
+buildGoModule rec {
+  pname = "docopts";
+  version = "0.6.4-with-no-mangle-double-dash";
+
+  src = fetchFromGitHub {
+    owner = "docopt";
+    repo = "docopts";
+    rev = "v${version}";
```
Comment (clebs) [inline pkgs/by-name/do/docopts/package.nix:13]:
```suggestion
    rev = "refs/tags/v${version}";
```

to avoid ambiguity

**Review by clebs — APPROVED** [review]
Result of `nixpkgs-review pr 318612` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>docopts</li>
  </ul>
</details>

Tested binary on: `x86_64-linux`, `aarch64-linux`, `aarch64-darwin`

**Comment by katexochen** [comment]
Not sure if @con-f-use is still maintaining this package, they are not in the NixOS org (anymore?) and thus weren't pinged by ofborg.

**Comment by con-f-use** [comment]
Hehe, yeah, I never was in the NixOS org, and that's the first time I've heard of this PR.

**Comment by katexochen** [comment]
> Hehe, yeah, I never was in the NixOS org. I'd like to be though. That's the first time I've heard of this PR, looks good.

@con-f-use I think https://github.com/NixOS/nixpkgs/issues/234293 is the right tread to ask for an (re-)invitation to the org. :)

**Pull request merged**